### ISSUE TITLE
Add support for the "textDocument/formatting" verb

### ISSUE THREAD

**Issue by vonlost** [apple/sourcekit-lsp#576]
Sorry if this request is covered elsewhere.

**Comment by ahoppen** [comment]
rdar://96159694

**Comment by siegel** [comment]
@fwcd which version of Xcode does this correspond to?

As of Xcode 13.4 (13F17a), `sourcekit-lsp` is not advertising support for `textDocument/formatting` in its capabilities:

```
{
    codeActionProvider =     {
    };
    colorProvider = 1;
    completionProvider =     {
        resolveProvider = 0;
        triggerCharacters =         (
            "."
        );
    };
    definitionProvider = 1;
    documentHighlightProvider = 1;
    documentSymbolProvider = 1;
    executeCommandProvider =     {
        commands =         (
            "semantic.refactor.command"
        );
    };
    foldingRangeProvider = 1;
    hoverProvider = 1;
    implementationProvider = 1;
    referencesProvider = 1;
    textDocumentSync =     {
        change = 2;
        openClose = 1;
        save =         {
            includeText = 0;
        };
        willSave = 1;
        willSaveWaitUntil = 0;
    };
    workspaceSymbolProvider = 1;
}
```

**Comment by fwcd** [comment]
Yes, IIUC the provider is not registered since the request is yet to be implemented (WIP at #220), only the request structures have been merged already.

**Comment by t-benze** [comment]
For anyone having the problem that sourcekit is not able to perform the code format action. If you happen to use neovim, there's one working solution using this [null-ls plugin](https://github.com/jose-elias-alvarez/null-ls.nvim/tree/main). 

There're a few steps to set up:

1. follow the [instruction](https://github.com/apple/swift-format#getting-swift-format) to install `swift-format`.
2. make sure the `swift-format` executable is locatable from $PATH. 
3. install the null-ls neovim plugin and [configure swift-format](https://github.com/jose-elias-alvarez/null-ls.nvim/blob/main/doc/BUILTINS.md#swift_format)
